core/src/main/java/com/linecorp/armeria/server/ServiceConfig.java

```diff
@@ -114,12 +112,11 @@ private ServiceConfig(@Nullable VirtualHost virtualHost, Route route,
         this.maxRequestLength = validateMaxRequestLength(maxRequestLength);
         this.verboseResponses = verboseResponses;
         this.accessLogWriter = requireNonNull(accessLogWriter, "accessLogWriter");
-        this.shutdownAccessLogWriterOnStop = shutdownAccessLogWriterOnStop;
         this.transientServiceOptions = requireNonNull(transientServiceOptions, "transientServiceOptions");
         this.blockingTaskExecutor = requireNonNull(blockingTaskExecutor, "blockingTaskExecutor");
-        this.shutdownBlockingTaskExecutorOnStop = shutdownBlockingTaskExecutorOnStop;
         this.successFunction = requireNonNull(successFunction, "successFunction");
         this.multipartUploadsLocation = multipartUploadsLocation;
+        this.shutdownSupports = ImmutableList.copyOf(requireNonNull(shutdownSupports, "shutdownSupports"));
```

Comment:
Q) Why is a copy made here? Would it help to have consistent behavior between `VirtualHost` and `ServiceConfig`? (Since `VirtualHost` doesn't seem to be copying `shutdownSupports`)